textEditorService.js

```diff
@@ -1,58 +1,53 @@
-const fs = require('fs');
-const {path: parsePath} = require('./parsers');
+const fs = require("fs");
+const { path: parsePath } = require("./parsers");
 
-module.exports = class FileTextEditor{
-    constructor(path) {
-        this.path = path;
-        this.fixedPath = parsePath(path);
-    }
+module.exports = class FileTextEditor {
+  constructor(path) {
+    this.path = path;
+    this.fixedPath = parsePath(path);
+  }
 
-    async createFile({content}){
-        try { 
-            await fs.writeFileSync(this.fixedPath, content);
-        }
-        catch (error) {
-            throw `Error creating file ${this.path}: ${error.message || JSON.stringify(error)}`
-        }
-        return `Created file ${this.path}.`;
+  async createFile({ content }) {
+    try {
+      await fs.writeFileSync(this.fixedPath, content);
+    } catch (error) {
+      throw new Error(`Error creating file ${this.path}: ${error.message || JSON.stringify(error)}`);
     }
+    return `Created file ${this.path}.`;
+  }
 
-    async appendToFile({content, createIfNeeded, returnContent}){
-        if (!fs.existsSync(this.fixedPath)) {
-            if (createIfNeeded) {
-                const result = await this.createFile({content});
-                return returnContent ? await fs.readFileSync(this.fixedPath, 'utf8') : result;
-            }
-            throw `File ${this.path} doesn't exist!`;
-        }
-        try { 
-            await fs.appendFileSync(this.fixedPath, content);
-        }
-        catch (error) {
-            throw `Error updating file ${this.path}: ${error.message || JSON.stringify(error)}`
-        }
-        return returnContent ? await fs.readFileSync(this.fixedPath, 'utf8') : `File ${this.path} updated.`;
+  async appendToFile({ content, createIfNeeded, returnContent }) {
+    if (!fs.existsSync(this.fixedPath)) {
+      if (createIfNeeded) {
+        const result = await this.createFile({ content });
+        return returnContent ? fs.promises.readFile(this.fixedPath, "utf8") : result;
+      }
+      throw new Error(`File ${this.path} doesn't exist!`);
     }
+    await fs.appendFileSync(this.fixedPath, content);
```

Comment:
Why no try/catch around that one?

Reply:
Because I lack the know-how to do that in a way that eslint is OK with. Care to tutor me?

Reply:
```suggestion
        try { 
            await fs.appendFileSync(this.fixedPath, content);
        }
        catch (error) {
            throw new Error(`Error updating file ${this.path}: ${error.message || JSON.stringify(error)}`);
        }
```
What about this one?